### PR TITLE
Implement SimplePolicyPlugin in Transaction.Verify

### PR DESCRIPTION
**Pull request by jseagrave21**
**What current issue(s) does this address, or what feature is it adding?**
- addresses https://github.com/CityOfZion/neo-python/issues/960
  - implements `SimplePolicyPlugin` when verifying a tx in the mempool

**How did you solve this problem?**
referenced logic used in #869 and logic I am using here https://github.com/jseagrave21/neo-python/blob/f75cb5dc58bedbfc12ea1a6afbca5332088f8ca6/neo/Core/TX/RawTransaction.py#L525-L541

**How did you make sure your solution works?**
`make test` and verification manually on my testnet node

**Are there any special changes in the code that we should be aware of?**
no

**Please check the following, if applicable:**

- [x] Did you add any tests?
- [x] Did you run `make lint`?
- [x] Did you run `make test`?
- [x] Are you making a PR to a feature branch or development rather than master?
- [x] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

**Comment by jseagrave21** [comment]
> The solution does not seem to take the following rule into account ([ref](https://neo.org/blog/details/4148)):
> 
> > Deploy and invoke smart contract with neo-cli v2.10.2 will cost 0.001 GAS as the starting fee.

This was clarified in #dev-general on Discord by @vncoelho that this was only introduced in neo-cli and not a limitation on Seed Nodes or Concensus Nodes. It was discussed further and removed here: https://github.com/neo-project/neo-cli/pull/378

**Comment by ixje** [comment]
I missed that. Thanks for the link!
Looks good then 👍 👍

**Comment by vncoelho** [comment]
The implementation looks precise and the formula correct.

**Comment by ixje** [comment]
thanks for the extra check @vncoelho

File: neo/Core/TX/Transaction.py [hunk 1]
```diff
@@ -594,6 +595,19 @@ def Verify(self, mempool):
...
         logger.debug("Verifying transaction: %s " % self.Hash.ToBytes())
 
+        # SimplePolicyPlugin
+        if self.Size() > self.MAX_TX_SIZE:
+            logger.debug(f'Maximum transaction size exceeded: {self.Size()} > {self.MAX_TX_SIZE}')
+            return False
+        fee = self.NetworkFee()
+        if self.Size() > settings.MAX_FREE_TX_SIZE and not self.Type == b'\x02':  # Claim Transactions are High Priority
+            req_fee = Fixed8.FromDecimal(settings.FEE_PER_EXTRA_BYTE * (self.Size() - settings.MAX_FREE_TX_SIZE))
+            if req_fee < settings.LOW_PRIORITY_THRESHOLD:
```
Comment (vncoelho) [inline neo/Core/TX/Transaction.py]: This was the trick point. Nice job!

**Comment by ixje** [comment]
100 points

**Comment by lllwvlvwlll** [comment]
https://neoscan.io/transaction/589328c622e101065f6717d61a332f93aac71cc68b665e194ef8c874a7e26669

**Pull request merged**